recommender/celery_app/celery_app/task1.py

```diff
@@ -0,0 +1,141 @@
+        return pickle.load(f)
+
+
+def get_menuItem(url_menuitem):
+    # get menuItem information
+    r = requests.get(url_menuitem).json()
+    item_id = {}
+    for i in range(len(r)):
+        item = r[i]
+        item_id[item['name']] = item['id']
+    print("MenuItem info extracted!")
+    return item_id
+
+
+def get_user_history(url_user, url_visit, url_order):
+    # get user ordering history
+    user_names = {}
+    user_item_list = {}
+    r = requests.get(url_user).json()
+    print(r.text)
+    for i in range(len(r)):
+        user = r[i]
+        user_names[user['id']] = user['firstName'] + \
+            "_"+user['lastName']
+        visits = user['visits']
+        user_item_list[user['id']] = []
+        for visit in visits:
+            order_id = requests.get(
+                url_visit+"/"+str(visit['id']))['orders'][0]['id']
+            items = requests.get(
+                url_order+"/"+str(order_id))['menuItems']
+            for item in items:
+                user_item_list[user['id']].append(item)
+    print("User ordering info extracted!")
+    return user_names, user_item_list
+
+
+def generate_review_data(data_name, user_item_list, item_id):
+    # create user ordering dataset according
+    with open(data_name, 'w') as dat:
+        dat.write("user,item,rating\n")
+        for user in user_item_list:
+            order_count = defaultdict(int)
+            for item in user_item_list[user]:
+                order_count[item['id']] += 1
+            factor = 1/sum(order_count.values())
+            for itemId in item_id.values():
+                order_count[itemId] *= factor
+                dat.write(
+                    ','.join([user, itemId, str(order_count[itemId]//0.01)])+'\n')
+
+
+def train_recommender(data_name, model_name):
+    reader = Reader(sep=',', rating_scale=(0, 99), skip_lines=1)
+    data = Dataset.load_from_file(data_name, reader=reader)
+    trainset = data.build_full_trainset()
+    algo = SVD()
+    algo.fit(trainset)
+    save_obj(algo, model_name)
+    print("model saved!!!")
+
+
+def load_recommender(model_name):
+    return load_obj(model_name)
+
+
+def recommend_all_users(user_names, item_id):
+    recommender = load_recommender("model")
+    recommended_item_IDs = defaultdict(list)
+    for user_id in user_names:
+        item_ratings = defaultdict(float)
+        for item in item_id:
+            item_ratings[item] = recommender.predict(
+                uid=user_id, iid=item_id[item]).est
+        sorted_item_ratings = sorted(
+            (item_ratings), key=lambda item: item_ratings[item], reverse=True)
+        for item in sorted_item_ratings:
+            recommended_item_IDs[user_id].append(item_id[item])
+    save_obj(recommended_item_IDs, "recommended_item_IDs")
+    print("recommended_item_IDs saved!!!")
+    return recommended_item_IDs
+
+
+@app.task
+def update_eta():
+    import time
+    url_order = "http://"+ENV_IP+"/api/v1/orders"
```

Comment:
You may just want to make a `base_url = "http://{}"/api/v1/".format(ENV_IP)` at the top and then just do `base_url+"/orders"`